### PR TITLE
make most functions work locally without a Node server (enhancement), i18n fixes, getScript failure recovery (bug)

### PR DESCRIPTION
**Pull request by duzc2**
Hello .  I am trying to make functions work locally without a server.
it works great now , but just lose some svg icons.

### CODE REVIEW

File: editor/extensions/ext-markers.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ svgEditor.addExtension("Markers", function(S) {
 	var marker_types = {
 		nomarker: {},  
 		leftarrow:  
-			{element:'path', attr:{d:'M0,50 L100,90 L70,50 L100,10 Z'}},
+			{element:'path', attr:{d:'M0,50 L100,90 L70,50 L100,10 Z'},icon:svgEditor.curConfig.extPath + 'markers-icons-leftarrow.png'},
```
Comment (codedread) [inline editor/extensions/ext-markers.js]: what is this about?

**Comment by brettz9** [comment]
I wonder if `$.ajax` in later jQuery versions allows for same-directory `file://` retrieval (perhaps with `isLocal`), so that this would be a non-issue if our jQuery version were upgraded as per #52 .... In the meantime (or instead), I think a better PR might simply replace our current use of `$.ajax` (once in `svg-editor.js` and once in `jQuery.svgIcons.js`) with our own `XMLHttpRequest`; in that way, the SVG icons should work locally without need for fallbacks to PNGs.

**Comment by brettz9** [comment]
Nevermind, it appears Chrome just won't allow `file:` locally for XHR.

**Comment by brettz9** [comment]
I've incorporated your fixes (i18n (which I confirmed) and images (which I expanded upon)). The images have also been optimized, and I've confirmed they are being applied locally when run from `file://`.

**Pull request closed without merging**